src/rust/zh.manhuagui/src/parser.rs

```diff
@@ -0,0 +1,329 @@
+];
+const FILTER_GENRE: [&str; 39] = [
+	"all",
+	"rexue",
+	"maoxian",
+	"mohuan",
+	"shengui",
+	"gaoxiao",
+	"mengxi",
+	"aiqing",
+	"kehuan",
+	"mofa",
+	"gedou",
+	"wuxia",
+	"jizhan",
+	"zhanzheng",
+	"jingji",
+	"tiyu",
+	"xiaoyuan",
+	"shenghuo",
+	"lizhi",
+	"lishi",
+	"weiniang",
+	"zhainan",
+	"funv",
+	"danmei",
+	"baihe",
+	"hougong",
+	"zhiyu",
+	"meishi",
+	"tuili",
+	"xuanyi",
+	"kongbu",
+	"sige",
+	"zhichang",
+	"zhentan",
+	"shehui",
+	"yinyue",
+	"wudao",
+	"zazhi",
+	"heidao",
+];
+const FILTER_AUDIENCE: [&str; 6] = [
+	"all", "shaonv", "shaonian", "qingnian", "ertong", "tongyong",
+];
+const FILTER_PROGRESS: [&str; 3] = ["all", "lianzai", "wanjie"];
+
+pub fn parse_home_page(html: Node) -> Result<MangaPageResult> {
+	let mut mangas: Vec<Manga> = Vec::new();
+
+	let ul = "#contList > li";
+	// let elements = html.select(ul).array();
+
+	for element in html.select(ul).array() {
+		let elem = element.as_node().unwrap();
+		let manga_id = elem
+			.select("a")
+			.attr("href")
+			.read()
+			.replace("/comic/", "")
+			.replace('/', "");
+		let title = elem.select("a").attr("title").read();
+		let manga = Manga {
+			id: manga_id.clone(),
+			cover: format!("https://cf.hamreus.com/cpic/b/{}.jpg", manga_id),
+			title,
+			author: String::new(),
+			artist: String::new(),
+			description: String::new(),
+			url: format!("{}/comic/{}", BASE_URL, manga_id), //`${this.baseUrl}/comic/${mangaId}`;,
+			categories: vec![],
+			status: MangaStatus::Completed,
+			nsfw: MangaContentRating::Safe,
+			viewer: MangaViewer::Scroll,
+		};
+		mangas.push(manga);
+	}
+
+	let mut has_next: bool = false;
+	for page in html.select("#AspNetPager1 > a").array() {
+		if page.as_node().unwrap().text().read() == "尾页" {
+			has_next = true;
+			break;
+		}
+	}
+
+	html.close();
+
+	Ok(MangaPageResult {
+		manga: mangas,
+		has_more: has_next,
+	})
+}
+
+pub fn parse_search_page(html: Node) -> Result<MangaPageResult> {
+	let mut mangas: Vec<Manga> = Vec::new();
+
+	let ul = ".cf > .book-cover > a";
+
+	for element in html.select(ul).array() {
+		let elem = element.as_node().unwrap();
+		let manga_id = elem
+			.attr("href")
+			.read()
+			.replace("/comic/", "")
+			.replace('/', "");
+		let title = elem.attr("title").read();
+		let manga = Manga {
+			id: manga_id.clone(),
+			cover: format!("https://cf.hamreus.com/cpic/b/{}.jpg", manga_id),
+			title,
+			author: String::new(),
+			artist: String::new(),
+			description: String::new(),
+			url: format!("{}/comic/{}", BASE_URL, manga_id), //`${this.baseUrl}/comic/${mangaId}`;,
+			categories: vec![],
+			status: MangaStatus::Completed,
+			nsfw: MangaContentRating::Safe,
+			viewer: MangaViewer::Scroll,
+		};
+		mangas.push(manga);
+	}
+
+	let mut has_next: bool = false;
+	for page in html.select("#AspNetPagerResult > a").array() {
+		if page.as_node().unwrap().text().read() == "尾页" {
+			has_next = true;
+			break;
+		}
+	}
+
+	html.close();
```

Comment:
you actually don't need to do this manually because rust automatically drops things that go out of scope.
```suggestion
```